### PR TITLE
[17409] Enable Type of care Agnostic

### PR DESCRIPTION
**Pull request by LanceGundersen**
## Description
Currently, we have a function in the location service file that fetches facilities from the two VATS criteria endpoints and combines them with the facility data from the facilities api. This function takes a type of care id and marks each facility as supporting or not supporting that type of care.

The criteria data we get back includes info about all types of care, we should be able to refactor the function in the location service file to not require a type of care id, and instead update each facility with a supported flag for each types of care and then update the redux and/or component logic to check those flags instead.

## Testing done
Local and Unit

## Screenshots
n/a

## Acceptance criteria
- [ ] No longer need to pass type of care to this service.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/new-appointment/components/VAFacilityPage/VAFacilityPageV2.multi.unit.spec.jsx [hunk 1]
```diff
@@ -456,7 +456,7 @@ describe('VAOS integration: VA flat facility page - multiple facilities', () =>
     ).to.exist;
   });
 
-  it('should show not supported message when direct is supported and not eligible, and requests are not supported', async () => {
+  it.skip('should show not supported message when direct is supported and not eligible, and requests are not supported', async () => {
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/VAFacilityPage/VAFacilityPageV2.multi.unit.spec.jsx:459]: Is this test not working?

Comment (LanceGundersen) [reply src/applications/vaos/tests/new-appointment/components/VAFacilityPage/VAFacilityPageV2.multi.unit.spec.jsx:459]: Oh no I missed it and so did the linter 🤔

Comment (LanceGundersen) [reply src/applications/vaos/tests/new-appointment/components/VAFacilityPage/VAFacilityPageV2.multi.unit.spec.jsx:459]: Fixed

File: src/applications/vaos/services/location/transformers.js [hunk 2]
```diff
@@ -257,23 +257,41 @@ export function transformATLASLocation(tasInfo) {
...
+    facility => `var${facility.id}` === id,
+  )?.requestSettings;
+  const facilityCoreSettings = directFacilities.find(
+    facility => `var${facility.id}` === id,
+  )?.coreSettings;
+
+  if (facilityRequestSettings) {
+    facilityRequestSettings.forEach(typeOfCare => {
+      requestSupported[typeOfCare.id] = !!typeOfCare.patientHistoryRequired;
+      return requestSupported;
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/services/location/transformers.js:283]: 🤦

**Review by jbalboni — APPROVED** [review]

**Pull request merged**